backend/src/router.go

```diff
@@ -349,3 +351,48 @@ func getProfile(c *gin.Context) {
 		"user": users[0],
 	})
 }
+
+
+//not all of this is needed to make an update, at least according to the guide. We shall see.
+type ProfileUpdateData struct { 
+	ID        uint   `json:"-"`
+	FirstName string `json:"firstname"`
+	LastName  string `json:"lastname"`
+	Email     string `json:"email"`
+	Phone     string `json:"phone"`
+}
+
+// ADD DESCRIPTION
+func patchProfile(c *gin.Context) {
+	
+	token, err := c.Cookie("jwt")
+	if err != nil {
+		c.JSON(401, gin.H{
+			"error": "Requires an authenticated user.",
+		})
+		return
+	}
+	
+	claims, err := ParseJWT(token)
+	if err != nil {
+		c.JSON(500, gin.H{
+			"error": "Unable to parse JWT: " + err.Error() + ".",
+		})
+		return
+	}
+	
+	var users []User
```

Comment:
Move declaration to directly before the `Find` call.

Reply:
Done :-)